### PR TITLE
Add GitHub Actions support

### PR DESCRIPTION
**Pull request by zooba**

### CODE REVIEW

**Comment by zooba** [comment]
TODO:
- [ ] Break some builds and check out error messages
- [ ] Set up path filters to only build on real changes

**Comment by zooba** [comment]
Real PR: https://github.com/python/cpython/pull/17594

File: .github/workflows/doc.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.7'
+        architecture: 'x64'
+    - name: 'Install build dependencies'
+      run: python -m pip install sphinx==2.2.0 blurb python-docs-theme
+    - name: 'Build documentation'
+      run: |
+        cd Doc
+        make check suspicious html PYTHON=python
```
Comment (methane) [inline .github/workflows/doc.yml]: How about upload the build html?

```
    - name: 'Build documentation'
      run: |
        cd Doc
        make check suspicious html PYTHON=python
        cd build
        tar cvzf html.tar.gz html
    - name: Upload
        uses: actions/upload-artifact@v1
        with:
          name: python-doc-html
          path: Doc/build/html.tar.gz
```

Comment (tirkarthi) [reply .github/workflows/doc.yml]: Are the uploaded artifacts hosted somewhere to preview docs rendering? There is netlify.toml to push to netlify but I guess the integration is still not complete. https://bugs.python.org/issue37860

Comment (methane) [reply .github/workflows/doc.yml]: > Are the uploaded artifacts hosted somewhere to preview docs rendering?

No, it is not hosted.  We can just download the tarball of the HTML.
Until we have netlify preview, the downloadable tarball is better than nothing.

**Pull request closed without merging**